### PR TITLE
Send active consumers with refresh token

### PR DESCRIPTION
**Pull request by nfmohit**
## Summary

Addresses issue:

- #5569 

## Relevant technical choices

This PR sends the new `active_consumers` with refresh token.

## PR Author Checklist

- [ ] My code is tested and passes existing unit tests.
- [ ] My code has an appropriate set of unit tests which all pass.
- [ ] My code is backward-compatible with WordPress 4.7 and PHP 5.6.
- [ ] My code follows the [WordPress](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) coding standards.
- [ ] My code has proper inline documentation.
- [x] I have added a QA Brief on the issue linked above.
- [x] I have signed the Contributor License Agreement (see <https://cla.developers.google.com/>).

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Code Reviewer Checklist

- [x] Run the code.
- [x] Ensure the acceptance criteria are satisfied.
- [x] Reassess the implementation with the IB.
- [x] Ensure no unrelated changes are included.
- [x] Ensure CI checks pass.
- [ ] Check Storybook where applicable.
- [x] Ensure there is a QA Brief.

## Merge Reviewer Checklist

- [x] Ensure the PR has the correct target branch.
- [x] Double-check that the PR is okay to be merged.
- [x] Ensure the corresponding issue has a ZenHub release assigned.
- [x] Add a changelog message to the issue.

### CODE REVIEW

File: includes/Core/Authentication/Clients/Google_Site_Kit_Client.php [hunk 1]
```diff
@@ -280,4 +282,26 @@ protected function handleAuthTokenErrorResponse( $error, array $data ) {
 		throw new Google_OAuth_Exception( $error );
 	}
 
+	/**
+	 * Create a default Google OAuth2 object.
+	 *
+	 * @return OAuth2 Crated OAuth2 instance.
```
Comment (techanvil) [inline includes/Core/Authentication/Clients/Google_Site_Kit_Client.php:288]: Typo
```suggestion
	 * @return OAuth2 Created OAuth2 instance.
```

File: includes/Core/Authentication/Clients/OAuth2.php [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+
+	/**
+	 * Overrides generateCredentialsRequest with additional parameters.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param array $extra_params Optional. Array of extra parameters to fetch with.
+	 * @return RequestInterface Token credentials request.
+	 */
+	public function generateCredentialsRequest( $extra_params = array() ) {
```
Comment (techanvil) [inline includes/Core/Authentication/Clients/OAuth2.php:35]: It would be nice to see some tests for this method...

**Review by techanvil — CHANGES_REQUESTED** [review]
Hey @nfmohit, this is looking great. I have left a couple of comments.

Additionally, with regard to this QAB point:

> - Open a non-Site Kit admin page (for example `Appearance > Themes`: /wp-admin/themes.php).

I am a bit confused because in my testing, I don't see the request to `https://sitekit.withgoogle.com/o/oauth2/token/ ` when I hit `Appearance > Themes`. However when I go to the Site Kit Dashboard I do see it, with `active_consumers` in the request payload as expected.

Not sure if I am missing something here or if the QAB needs a tweak - please take a look.

File: includes/Core/Authentication/Clients/Google_Site_Kit_Client.php (hunk 1 above)
Comment (nfmohit) [reply includes/Core/Authentication/Clients/Google_Site_Kit_Client.php]: Fixed, thank you for spotting!

File: includes/Core/Authentication/Clients/OAuth2.php (hunk 2 above)
Comment (nfmohit) [reply includes/Core/Authentication/Clients/OAuth2.php:35]: Added a test case, thank you!

**Comment by nfmohit** [comment]
Thank you for the kind review, @techanvil!

I have addressed the comments.

As for the QAB, I have updated it to go to the Site Kit Dashboard.

I have also fixed the merge conflicts.

Back to you now.

File: tests/phpunit/integration/Core/Authentication/Clients/OAuth2Test.php [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+ */
+class OAuth2Test extends TestCase {
+
+	/**
+	 * Parse request body stream.
+	 *
+	 * @param RequestInterface $request The request to parse body for.
+	 * @return array The parsed body.
+	 */
+	private static function getParsedRequestBody( $request ) {
```
Comment (techanvil) [inline tests/phpunit/integration/Core/Authentication/Clients/OAuth2Test.php:31]: Sorry I missed it the first time round, but please can you change this method name to snake case in line with WordPress PHP naming conventions (I know it's a bit of a mix in this PR, with good reason).

```suggestion
	private static function get_parsed_request_body( $request ) {
```

**Review by techanvil — CHANGES_REQUESTED** [review]
Thanks @nfmohit! Just one small tweak that I managed to miss the first time around, and this looks good to go :+1:

File: tests/phpunit/integration/Core/Authentication/Clients/OAuth2Test.php (hunk 3 above)
Comment (nfmohit) [reply tests/phpunit/integration/Core/Authentication/Clients/OAuth2Test.php]: Uh oh, sorry for mixing it up. Updated, thank you!

**Review by techanvil — APPROVED** [review]
LGTM, nice one @nfmohit!

**Pull request merged**